Tests/Rules/ReviewUnusedParameter.tests.ps1

```diff
@@ -57,7 +57,13 @@ Describe "ReviewUnusedParameter" {
         }
 
         It "has no violations when parameter is called in child scope" -skip {
-            $ScriptDefinition = 'function Param { param ($Param1) function Child { $Param1 } }'
+            $ScriptDefinition = 'function foo { param ($Param1) function Child { $Param1 } }'
```

Comment:
I would think of this as a violation. Since PowerShell has dynamic, rather than lexical, scope, `Child`'s `$Param1` reference is not guaranteed to be `foo`'s `$Param1` parameter.

Reply:
I'd rather avoid violations that are going to be false positives in most cases in order to avoid similar problems to `PSUseDeclaredVarsMoreThanAssignments`

Reply:
I think fair enough we want to not emit when at the boundary of our heuristic. Ideally we'd change this to an actual false case, but it's not that important.